src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -974,10 +970,10 @@ private void EncodePixels<TPixel>(Image<TPixel> pixels, IndexedImageFrame<TPixel
 
             for (int y = 0; y < this.height; y++)
             {
-                IManagedByteBuffer r = this.EncodePixelRow(pixels.GetPixelRowSpan(y), quantized, y);
-                deflateStream.Write(r.Array, 0, resultLength);
+                IMemoryOwner<byte> r = this.EncodePixelRow(pixels.GetPixelRowSpan(y), quantized, y);
```

Comment:
Unless I'm missing something, looks like we are leaking here:

```suggestion
                using IMemoryOwner<byte> r = this.EncodePixelRow(pixels.GetPixelRowSpan(y), quantized, y);
```

Reply:
Yeah, the code is odd. It's actually returning one of too many different buffers. Have refactored the encoder to simplify things.